src/shapes/TextPath.ts

```diff
@@ -105,6 +107,21 @@ export class TextPath extends Shape<TextPathConfig> {
     this._setTextData();
   }
 
+  _readDataAttribute() {
+    this.dataArray = Path.parsePathData(this.attrs.data);
+    // in case document is not defined (server side rendering)
+    // use the KonvaJs Path class instead of the browser's native SVGPathElement
+    if (typeof window !== 'undefined' && document) {
+      this.path = document.createElementNS(
+        'http://www.w3.org/2000/svg',
+        'path'
+      ) as SVGPathElement;
+      this.path.setAttribute('d', this.attrs.data);
+    } else {
+      this.path = new Path({ data: this.attrs.data }) as Path;
```

Comment:
I would think how to avoid creating an instance of the path. Probably we can somehow just use methods from it directly. It will save memory a bit. But more important, shapes must be properly destroyed. So instead of thinking about how to destroy shapes, it may be better not to create them at all.

Reply:
Good point. All I need in this PR is `getLength` calculation and get `getPointAtLength`.
The `getLength` can be replaced with: 
```js
this.pathLength = 0;
  for (var i = 0; i < this.dataArray.length; ++i) {
    this.pathLength += this.dataArray[i].pathLength;
}
```
Because this is how `Path` calculates it in `constructor`. 

`getPointAtLength` -  I think it is possible to create a new static method that accepts `length` and `dataArray` as a second parameter. Then use a static method in TextPath (and in `getPointAtLength` too). It should do the trick, right?

Reply:
Sounds good to me.

Reply:
@lavrton WDYT about this approach?